generators/client/templates/react/src/main/webapp/app/modules/account/register/register.tsx

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react';
+import { Translate, translate } from 'react-jhipster';
+import { connect } from 'react-redux';
+
+import { AvForm, AvField } from 'availity-reactstrap-validation';
+import { Row, Col, Alert, Button } from 'reactstrap';
+import PasswordStrengthBar from '../password/password-strength-bar';
+
+import { handleRegister, reset } from '../../../reducers/register';
+
+export const mainErrorMessages = {
+  failed: (
+    <Translate contentKey="register.messages.error.fail">
+      <strong>Registration failed!</strong> Please try again later.
+    </Translate>
+  ),
+  userexists: (
+    <Translate contentKey="register.messages.error.userexists">
+      <strong>Login name already registered!</strong> Please choose another one.
+    </Translate>
+  ),
+  emailexists: (
+    <Translate contentKey="register.messages.error.emailexists">
+      <strong>Email is already in use!</strong> Please choose another one.
+    </Translate>
+  )
+};
+
+export interface IRegisterProps {
+  handleRegister: Function;
+  reset: Function;
+  registrationSuccess: boolean;
+  registrationFailure: boolean;
+  errorMessage: string;
+}
+
+export interface IRegisterState {
+  password: string;
+}
+
+export class RegisterPage extends React.Component<IRegisterProps, IRegisterState> {
+  constructor(props) {
```

Comment:
No need of constructor do this instead. I'm planning to change this for existing components as well
```
state: IRegisterState = {
  password: ''
}
```